### ISSUE TITLE
Empty integer enum is treated as zero?

### ISSUE THREAD

**Issue by vebjornjr** [json-editor/json-editor#1175]
### General information

- json-editor version: 2.8.0

### Expected behavior

With a JSON schema with a property as
```
"OutputPwmId": {
    "type": "integer",
    "default": 1,
    "enum": [
      -1,
      0,
      1,
      2
    ]
  },
```
, I expect the error message _"Value must be one of the enumerated values."_ if no value is selected from the drop-down list. If I remove 0 as one of enums, then it works as expected.

### Actual behavior

No error message is emitted and the JSON object submitted has 0 as the value of OutputPwmId.

**Comment by germanbisurgi** [comment]
I agree, the value should be undefined. We eill take a kook at this. A temp workaround would be to make that editor required so the empty value can't be selected.

**Comment by schmunk42** [comment]
@vebjornjr This is a pretty though edge-case regarding how things work under the hood.
Actually, when selecting the empty value in the dropdown, the value should be removed in the output JSON, but I am not sure if this could be implemented easily.

We have the `show_opt_in` feature for these use-cases, but I noticed there's also an empty value - I think this should be fixed. @germanbisurgi 

![Bildschirmfoto vom 2022-07-05 12-58-11](https://user-images.githubusercontent.com/649031/177313158-1fa2e6bf-5718-41ff-a37b-15b0192e8c1b.png)

**Comment by germanbisurgi** [comment]
fixed in #1269
